lib/avrora/client.ex

```diff
@@ -93,16 +93,16 @@ defmodule Avrora.Client do
     config =
       quote do
         defmodule Config do
+          @dialyzer {:no_match, [schemas_path: 0]}
           @moduledoc false
-          @opts unquote(opts)
 
-          import Keyword, only: [get: 3]
+          @opts unquote(opts)
+          @otp_app Keyword.get(@opts, :otp_app)
 
           def schemas_path do
             path = get(@opts, :schemas_path, "./priv/schemas")
-            otp_app = get(@opts, :otp_app, nil)
 
-            if is_nil(otp_app), do: Path.expand(path), else: Application.app_dir(otp_app, path)
+            if is_nil(@otp_app), do: Path.expand(path), else: Application.app_dir(@otp_app, path)
```

Comment:
as things currently stand, there is no way to fully configure the schemas path. 
For example, in tests I would love to have schemas in tests/fixtures/avro_schemas/something.
If I do not use otp_app, then I can do so. But as soon as I introduce otp_app to be able to configure everything else at runtime, the schemas path gets resolved to the app_dir, which makes impossible to fully control it.

In my reference PR (https://github.com/Strech/avrora/pull/91/commits/8b5b25cfc1ff922d70d6114b5afe808112d56a66) I only expanded the path to Application.app_dir if the schemas_path was not configured at runtime. Which I think it will make the change backwards compatible.

Is that something you are willing to introduce?

Reply:
Thanks for your comment! I understand your concerns.

I would suggest this approach to gain full control over the configuration

1. Use [Mox library](https://github.com/dashbitco/mox)
2. Add your _"test"_ configuration [like this one](https://github.com/Strech/avrora/blob/master/test/support/config.ex)
3. [Redefine in tests](https://github.com/Strech/avrora/blob/master/test/support/config.ex#L25) any option you like

All you will need later is to define a mock and allow your test config to be used as default when nothing is defined

```elixir
Mox.defmock(Avrora.ConfigMock, for: Avrora.Config)
Mox.stub_with(Avrora.ConfigMock, Support.Config) # Support.Config is your "test" configuration
```

And then in the tests, you can simply do

```elixir
stub(Avrora.ConfigMock, :schemas_path, fn -> "./my/custom/path" end)
```

Reply:
And the reason why `app_dir` is used when `otp_app` is set and not just a value – due to possible root path change if you later will do `mix release`.

I wish to add that specially crafted end-to-end test suite I have locally, but need to figure out few things

Reply:
right, i understand the reason of the expansion of the schemas path.
But I think that was required when there was no way to configure things at runtime. now that we can, i see it more like a limitation on the configurability than a feature.
But if you want to keep it this way, then all good, i'll have to introduce mocks to test the behavior i need to test.

regardless, thank you very much for your time!!!

Reply:
>  But I think that was required when there was no way to configure things at runtime. now that we can, i see it more like a limitation on the configurability than a feature.

Hmmm, I get what you mean. I would say – in this case when we have `otp_app` it's the default behavior, like `Path.expand` when we don't have it set. You can think about it as – we expand the path to absolute via the best option we have.

In addition – with `Application.put_env(:otp_app, Client, config: Config)` you can avoid `app_dir` in the tests (also you don't need Mox for that if all your tests will have same test config or if you fine to use `put_env` everywhere, which is meh)

Reply:
awesome! i will explore the options. I have enough workarounds to find the best one for me.